apache/trafficcontrol
# Have the asf-ci user create Pull Requests for GitHub Actions

This PR makes Pull Requests created by our GitHub Actions use an API token created by the @asf-ci account (secret for the token requested in [INFRA-23048](https://issues.apache.org/jira/browse/INFRA-23048)). This makes our GitHub Actions actually run on these GHA-created PRs, since PRs authored by the @github-actions user do not trigger any GHAs. The branches and commits are still built using the token from @github-actions.
<hr/>

## Which Traffic Control components are affected by this PR?

- Automation - GitHub Actions

## What is the best way to verify this PR?

1. Push the [`zrhoffman:asf-ci-token`](https://github.com/zrhoffman/trafficcontrol/commits/asf-ci-token) to the `master` branch of your own fork
2. Cherry-pick 162af3f2bd onto the `master` branch of your fork, which downgrades the Go version to 1.17.7
3. Go to https://github.com/settings/tokens
4. Generate a token then has the *public_repo* scope enabled
5. Go to https://github.com/<Your Username\>/trafficcontrol/settings/secrets/actions
6. Create a new Repository Secret named `ASFCI_TOKEN` which contains that token
7. Manually trigger forkflow runs:
   - https://github.com/<Your Username\>/trafficcontrol/actions/workflows/assign-triage-role.yml
   - https://github.com/<Your Username\>/trafficcontrol/actions/workflows/pr-to-update-go.yml
8. Go to https://github.com/<Your Username\>/trafficcontrol/pulls and verify that the generated PRs are authored by you, not GitHub Actions

Or check these successful runs:
[*Is it time to update Go?*](https://github.com/zrhoffman/trafficcontrol/actions/runs/2092014423) run
[*Assign Triage Role*](https://github.com/zrhoffman/trafficcontrol/actions/runs/2092015043) run

## PR submission checklist
